src/System.Private.ServiceModel/src/System/ServiceModel/Description/CustomAttributeProvider.cs

```diff
@@ -83,7 +84,10 @@ public object[] GetCustomAttributes(Type attributeType, bool inherit)
                 case AttributeProviderType.MemberInfo:
                     return this.MemberInfo.GetCustomAttributes(attributeType, inherit).ToArray();
                 case AttributeProviderType.ParameterInfo:
-                    return this.ParameterInfo.GetCustomAttributes(attributeType, inherit).ToArray();
+                    //GetCustomAttributes could return null instead of empty collection for unknown reason here, workaround the issue by explicitly checking the null
+                    IEnumerable<Attribute> customAttributes = null;
```

Comment:
In this case:
The attributeType is MessageParameterAttribute, derives from Attribute.
The Parameter Info is System.ServiceModel.Channels.Message, without any custom attributes defined.

We expect System.Reflection.RuntimeParameterInfo.GetCustomAttribute returns an empty collection per it's definition about return value:
        // Returns:
        //     A collection of the custom attributes that are applied to element and that match
        //     attributeType, or an empty collection if no such attributes exist.

It should be a System.Reflection bug. I will attempt a standalone simple repro for System.Reflection team.